### ISSUE TITLE
Libre Alarms Using Non Calibrated Data

### ISSUE THREAD

**Issue by kylmcw** [nightscout/Trio#167]
**Describe the bug**
Using Libre, when a high or low notification is sounded, the app appears to be using non calibrated data to trigger the alarm, despite being lower (in this case) than the alarm trigger point.

**To Reproduce**
1. Calibrate Libre sensor to a value below default 
2. Observe when default value in GCM settings exceeds threshold, but calibrated value is below.
3. Observe high alert 

**Expected behavior**
High alert only applies to calibrated values.

**Screenshots**

![image](https://github.com/nightscout/Trio/assets/156543991/e408ca46-a0fb-4faa-93d0-acc739fd27e0)
![image](https://github.com/nightscout/Trio/assets/156543991/00f7d4c2-03e8-4148-b76d-e147fcd44e82)
![image](https://github.com/nightscout/Trio/assets/156543991/75a2f13c-3c3e-4d9a-a27b-654e4861df6b)
**Smartphone (please complete the following information):** iPhone 15 Pro

**Setup Information (please complete the following information):**
* Pump type - DASH
* CGM type and CGM app - Freestyle Libre 2, Libre App (don't use)
* Trio version, branch and git reference (see Trio Settings) Open-iAPS Alpha c8ceccc

**Additional context**

**Comment by bjornoleh** [comment]
Tagging @dabear in case this is not on your radar already.

**Comment by dabear** [comment]
I am not certain I understand this problem. Libretransmitter has its own experimental calibration feature. Libretransmitters alarm feature does work on top of this calibration. allowsEditingFactoryCalibrationData must be set to true though (it's experimental so not something that should be set to true by default).

If trio implements calibration on top of the value returned by Libretransmitter, Libretransmitter would not know about or use that for alarming. If Trio modifies the glucose values returned,  it should also be responsible for sending out any alarms rather than letting Libretransmitter do it

**Comment by bjornoleh** [comment]
Ok, thanks for the info, I believe Trio has its own calibration on top. Maybe the Libre alarms should be disabled when libretransmitter is used in Trio?

Tagging @kskandis , whom I believe is working on alarms/alerts in Trio?

**Comment by dabear** [comment]
If needed, I can add a feature flag to libretransmitter to programatically disable alarms, both from ui and the alarm handling code itself

**Comment by bjornoleh** [comment]
I believe Loop now also supports a cheaper Dexcom targeted at T2 without alarms in its accompanying app. Hence this discussion will probably reemerge. Adding the alarms at the CGM manager level sounds a little like duplicating of work across CGM managers.

**Comment by marionbarker** [comment]
Pete did not accept the pull request for Stelo support into Loop yet. He is wearing one himself but was concerned about alarms.

**Comment by kskandis** [comment]
> Ok, thanks for the info, I believe Trio has its own calibration on top. Maybe the Libre alarms should be disabled when libretransmitter is used in Trio? And replaced with Trio specific alarms perhaps?
> 
> Tagging @kskandis , whom I believe is working on alarms/alerts in Trio?

Yes, agree, Libre plugin has its own set of notifications that are enabled in the plugin by the user in NotificationsSettingsView, saved to UserDefaults, and based on those settings, Libre will send APN in NotificationsHelper.  This is similar to what Trio does.  Trio ,of course, already has High/Low BG notifications if the user enables them in Settings - Notifications - Glucose Notifications.  The other Libre notifications are Libre specific which Trio currently has no way of accessing that data.
